### PR TITLE
Central coach reports

### PR DESCRIPTION
**Pull request by rtibbles**
This PR makes changes to the coach reports that leave their functionality unchanged on the distributed server, but make them compatible with the Central Server code.

Summary of changes:
* Do not create dummy Django models for Groups 'All' and 'Ungrouped', instead dynamically generate those options on the client side.
* Use CENTRAL_SERVER aware permissions (borrowing from existing CENTRAL_SERVER compatible permissions code) for student progress reports.

### CODE REVIEW

**Comment by benjaoming** [comment]
@rtibbles there's a test failure

**Comment by rtibbles** [comment]
Oh dear. Thanks for pointing this out, @benjaoming it slipped off my radar.

File: kalite/coachreports/static/js/coachreports/coach_reports/views.js [hunk 1]
```diff
@@ -473,7 +474,10 @@ var FacilitySelectView = Backbone.View.extend({
             facilities: this.facility_list.toJSON(),
             selected: this.model.get("facility")
         }));
-        this.facility_changed();
+        var id = this.model.get("facility");
+        this.model.set({
+            "facility_name": this.facility_list.find(function(model){ return model.get("id") === id;}).get("name")
```
Comment (jamalex) [inline kalite/coachreports/static/js/coachreports/coach_reports/views.js]: May be enough: `this.facility_list.get(model.get("id")).get("name")`

File: kalite/coachreports/static/js/coachreports/coach_reports/views.js [hunk 2]
```diff
@@ -513,11 +530,17 @@ var GroupSelectView = Backbone.View.extend({
...
         }
+
         this.$el.html(this.template({
             groups: this.group_list.toJSON(),
             selected: this.model.get("group")
         }));
 
+        var id = this.model.get("group");
+        this.model.set({
+            "group_name": this.group_list.find(function(model){ return model.get("id") === id;}).get("name")
```
Comment (jamalex) [inline kalite/coachreports/static/js/coachreports/coach_reports/views.js]: Here too.

**Pull request merged**